src/main/kotlin/no/fdk/fdk_public_service_harvester/adapter/ServicesAdapter.kt

```diff
@@ -15,24 +15,25 @@ private const val TEN_MINUTES = 600000
 @Service
 class ServicesAdapter {
 
-    fun fetchServices(source: HarvestDataSource): String? {
+    fun fetchServices(source: HarvestDataSource): String {
         with(URL(source.url).openConnection() as HttpURLConnection) {
             try {
                 setRequestProperty("Accept", source.acceptHeaderValue)
                 connectTimeout = TEN_MINUTES
                 readTimeout = TEN_MINUTES
 
                 return if (responseCode != HttpStatus.OK.value()) {
-                    LOGGER.error("${source.url} responded with ${responseCode}, harvest will be aborted", HarvestException(source.url ?: "undefined"))
-                    null
+                    val exception = HarvestException("${source.url} responded with ${responseCode}, harvest will be aborted")
+                    LOGGER.error("${source.url} responded with ${responseCode}, harvest will be aborted", exception)
```

Comment:
Kjenner ikke til loggeren, men kommer feilmeldingen dobbelt opp i loggen nå? (siden både teksten og exception med teksten logges)

Reply:
Vel, ja. Etter som jeg husker ligger message fra selve feilmeldingen på slutten av en lang stacktrace. Vi legger til exception i loggen for å få med en stacktrace, det er det som må til for at error-reporting i GCP plukker opp feilen.

Ville ha med en mer beskrivende tekst i selve exception, fordi det er det som blir lagt til i høsterapporten.

Det som potensielt er verre er at feilen blir logget to ganger, en gang her og på ny når den blir plukket opp av try-catch i høsteprosessen. Skal gå gjennom feilmeldingene som har dukket opp i staging-miljøet i morgen. blir sikkert litt refaktorering etterpå.